algorithms/src/snark/varuna/tests.rs

```diff
@@ -524,3 +524,293 @@ mod varuna_hiding {
+    fn create_test_vector(folder: &str, file: &str, data: &str) {
+        // Construct the path for the test folder.
+        let mut path = resources_path();
+        path.push(folder);
+
+        // Create the test folder, if it does not exist.
+        if !path.exists() {
+            fs::create_dir(&path).unwrap_or_else(|_| panic!("Failed to create test folder: {path:?}"));
+        }
+
+        // Construct the path for the test file.
+        path.push(file);
+        path.set_extension("txt");
+
+        // Create the test file, if it does not exist.
+        if !path.exists() {
+            fs::File::create(&path).unwrap_or_else(|_| panic!("Failed to create file: {path:?}"));
+        }
+
+        fs::write(&path, data).unwrap_or_else(|_| panic!("Failed to write to file: {:?}", path));
+    }
+
+    #[test]
+    fn test_proving_system_with_test_vectors() {
+        run_prover_sage_vectors(false);
+    }
+
+    #[test]
+    #[ignore]
+    fn create_prover_test_vectors() {
+        run_prover_sage_vectors(true);
+    }
+
+    // TODO: make macros to use different hiding modes
+    fn run_prover_sage_vectors(create_test_vectors: bool) {
+        // Get the circuit instance from the test vector.
+        let instance_path = "src/snark/varuna/resources/instance.input";
+        let instance_file = fs::read_to_string(instance_path).expect("Could not read the file");
+        let mut instance = Vec::new();
+        for line in instance_file.lines() {
+            instance.push(line.to_string())
+        }
+
+        // Initialize challenges from file.
+        let challenges_path = "src/snark/varuna/resources/challenges.input";
+        let challenges_file = fs::read_to_string(challenges_path).expect("Could not read the file");
+        let mut challenges = Vec::new();
+        for line in challenges_file.lines() {
+            challenges.push(line)
+        }
+        let (alpha, _eta_a, eta_b, eta_c, beta, delta_a, delta_b, delta_c, _gamma) = (
+            Fr::from_str(challenges[0]).unwrap(),
+            Fr::from_str(challenges[1]).unwrap(),
+            Fr::from_str(challenges[2]).unwrap(),
+            Fr::from_str(challenges[3]).unwrap(),
+            Fr::from_str(challenges[4]).unwrap(),
+            vec![Fr::from_str(challenges[5]).unwrap()],
+            vec![Fr::from_str(challenges[6]).unwrap()],
+            vec![Fr::from_str(challenges[7]).unwrap()],
+            Fr::from_str(challenges[8]).unwrap(),
+        );
+        let circuit_combiner = Fr::one();
+        let instance_combiners = vec![Fr::one()];
+
+        let mut random_state = vec![0u64; 100];
+        let add_f_to_state = |s: &mut Vec<u64>, f: Fp256<FrParameters>| {
+            // Fp384 field elements sample 6 u64 values in total.
+            for i in 0..f.0.0.len() {
+                s.push(f.0.0[f.0.0.len() - 1 - i]);
+            }
+        };
+        for witness in instance[15].split(',') {
```

Comment:
I added a TODO that we can test the other instance values in the future: https://github.com/orgs/AleoHQ/projects/13?pane=issue&itemId=40044078